runtime/parachains/src/inclusion/mod.rs

```diff
@@ -932,6 +1166,26 @@ impl<BlockNumber> AcceptanceCheckErr<BlockNumber> {
 	}
 }
 
+impl<T: Config> OnQueueChanged<AggregateMessageOrigin> for Pallet<T> {
+	// Write back the remaining queue capacity into `relay_dispatch_queue_remaining_capacity`.
+	fn on_queue_changed(origin: AggregateMessageOrigin, count: u64, size: u64) {
+		let para = match origin {
+			AggregateMessageOrigin::Ump(p) => p,
+		};
+		// TODO maybe migrate this to u64
+		let (count, size) = (count.saturated_into(), size.saturated_into());
```

Comment:
They seem to already be `u64`...

Reply:
Yes, the `saturated_into` is downcasting to `u32`. There was an audit issue about effectively exhausting the `u32`, but I would consider it outside of the scope for this MR.